lib/src/pure_http/http_interceptor.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:convert';
+import 'dart:async';
+import 'package:http/src/base_request.dart';
+import 'package:http/src/client.dart';
+import 'package:http/src/base_client.dart';
+import 'package:http/src/streamed_response.dart';
+import 'package:http_mock_adapter/src/Exceptions.dart';
+import './http_response.dart';
+
+@Deprecated("This adapter class is an experimental feature")
```

Comment:
Deprecation is good idea, but i think this files should be moved and saved in another branch.